src/theory/datatypes/sygus_datatype_utils.cpp

```diff
@@ -359,6 +370,27 @@ Node sygusToBuiltin(Node n, bool isExternal)
           }
         }
       }
+      else if (cur.getType().isSygusDatatype())
+      {
+        Assert (cur.isVar());
+        if (cur.hasAttribute(SygusToBuiltinVarAttribute()))
+        {
+          // use the previously constructed variable for it
+          visited[cur] = cur.getAttribute(SygusToBuiltinVarAttribute());
+        }
+        else
+        {
+          std::stringstream ss;
+          ss << cur;
+          const DType& dt = cur.getType().getDType();
+          // make a fresh variable
+          NodeManager * nm = NodeManager::currentNM();
```

Comment:
```suggestion
          NodeManager* nm = NodeManager::currentNM();
```
Moreover move this to outside the loop.

Reply:
Its intentionally here for performance reasons (this is partly the point of the comment mentioned above).